src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrSyncToolingE2ETest.cs

```diff
@@ -0,0 +1,624 @@
+
+        Directory.CreateDirectory(_tmpPath);
+        CopyDirectory(_commonVmrPath, _vmrPath);
+        CopyDirectory(_commonPrivateRepoPath, _privateRepoPath);
+        CopyDirectory(_commonDependencyRepoPath, _dependencyRepoPath);
+
+        File.WriteAllText(
+            _vmrPath / "src" / "source-mappings.json",
+            string.Format(_sourceMappingsTemplate,
+            _privateRepoPath.Path.Replace("\\", "\\\\"),
+            _dependencyRepoPath.Path.Replace("\\", "\\\\")));
+
+        await CommitAll(_vmrPath, "Add source mappings");
+
+        var versionDetails = 
+            @"<?xml version=""1.0"" encoding=""utf-8""?>
+            <Dependencies>
+            <ProductDependencies>
+            </ProductDependencies>
+            <ToolsetDependencies>
+            <Dependency Name=""Dependency"" Version=""8.0.0"">
+            <Uri>{0}</Uri>
+            <Sha>{1}</Sha>
+            <SourceBuild RepoName=""dependency"" ManagedOnly=""true"" />
+            </Dependency>
+            </ToolsetDependencies>
+            </Dependencies>";
+
+        var commit = await GetRepoLastCommit(_dependencyRepoPath);
+        File.WriteAllText(_privateRepoPath / "eng" / _versionDetailsName, string.Format(versionDetails, new object[] { _dependencyRepoPath.Path, commit }));
+        await CommitAll(_privateRepoPath, "Add version details file");
+    }
+
+    [TearDown]
+    public void DeleteCurrentTestDirectory()
+    {
+        try
+        {
+            if (_currentTestDirectory is not null)
+            {
+                DeleteDirectory(_currentTestDirectory.ToString());
+            }
+        }
+        catch
+        {
+            // Ignore
+        }
+    }
+
+    [Test]
+    public async Task RepoIsInitializedTest()
+    {
+        var testRepoFilePath = _vmrPath / "src" / "test-repo" / "test-repo-file.txt";
+        var dependencyFilePath = _vmrPath / "src" / "dependency" / "dependencyFile.txt";
+
+        var commit = await GetRepoLastCommit(_privateRepoPath);
+        await CallDarcInitialize(_vmrPath, _tmpPath, "test-repo", commit);
+
+        var expectedFilesFromRepos = new List<LocalPath> 
+        {
+            testRepoFilePath, 
+            dependencyFilePath
+        };
+
+        var expectedFiles = GetExpectedFilesInVmr(
+            _vmrPath,
+            new[] { "test-repo", "dependency" },
+            expectedFilesFromRepos
+        );
+        
+        CheckDirectoryContents(_vmrPath, expectedFiles);
+        CheckFileContents(testRepoFilePath, "Test repo file");
+        CheckFileContents(dependencyFilePath, "File in the dependency repo");
+        await CheckAllIsCommited(_vmrPath);
+    }
+
+    [Test]
+    public async Task FileChangesAreSyncedTest()
+    {
+        var testRepoFilePath = _vmrPath / "src" / "test-repo" / "test-repo-file.txt";
+        var dependencyFilePath = _vmrPath / "src" / "dependency" / "dependencyFile.txt";
+
+        await RepoIsInitializedTest();
```

Comment:
It is a bit unusual to call a test method from another test method but also I am wondering why not?

Might be worth refactoring this out onto a non-Test method to make it clear something else depends on this and then only call this from the Initialized test method. Do this to prevent people changing the initialized test and influencing other tests unknowingly?